### ISSUE TITLE
Allow ariaRegion in Choice components to be toggled via prop

### ISSUE THREAD

**Issue by hbillings** [CMSgov/design-system#484]
## Background
Over on the [CMS HITECH APD project](https://github.com/18F/cms-hitech-apd/), we ran into [a rather hilarious issue](https://github.com/18F/cms-hitech-apd/pull/1731) where, immediately upon page load, our app announces all non-empty ARIA regions to screen readers. (I believe it has something to do with our wrapper architecture not being seen as a new page load, but as a set of new components mounted into an existing page.) 

We managed to fix this for our own components, but the CMS Design System's Choice component with conditional children wraps ARIA regions around those children by default, causing [this bug.](https://github.com/18F/cms-hitech-apd/issues/1743) 

## Our temporary workaround
We fixed this in our project by [writing a wrapper](https://github.com/18F/cms-hitech-apd/pull/1760) around the Choice component that brute-force removes the ARIA region (content still is announced to a screen reader, at least in Safari with VoiceOver).

## Proposed change
The wrapper solution is clunky and introduces unnecessary overhead. It would be really nice if we could just pass something like an `ariaRegion={false}` prop to the component to toggle it.

**Comment by line47** [comment]
Hey @davidakennedy could you take a look at this when you have a moment 👍 >?

**Comment by davidakennedy** [comment]
Hey @hbillings! Thanks for creating this issue. I'm researching it a bit, and want to make sure I understand.

By passing that additional prop, you'd be able to remove [this bit of code](https://github.com/CMSgov/design-system/blob/master/packages/core/src/components/ChoiceList/Choice.jsx#L108-L113), and handle it on your own? Or am I misunderstanding?

Thanks!

**Comment by davidakennedy** [comment]
Hey @hbillings – we chatted a bit more about this issue. If you can, it would be helpful if we could see a demo of this in action so we can get a better sense of what's causing the issue. Thanks!

**Comment by hbillings** [comment]
@davidakennedy Hi! Apologies for the delay in responding to this. Your assumption is correct! (Except I don't need to remove the div, just the aria attributes.) In our SPA, navigation to a new page reads the `aria-live` attributes aloud on page load (before there have been any changes), because it thinks that those elements being mounted is a change to the DOM that it must announce to a screen reader. (The wrapper I wrote for our project)[https://github.com/18F/cms-hitech-apd/pull/1760] shows you exactly which attributes I had to remove, and the original issue I linked to describes the problem we were having in greater detail. (Unfortunately, I can't really demo it for you since we've fixed the issue on our side!)

**Comment by bernardwang** [comment]
It looks like the `aria-live` is only applied when `<Choice>` has a `checkedChildren` element. This way when the user selects a choice that triggers `checkedChildren` to show, we announce the new child element.

I haven't exactly been able to reproduce this issue, probably of the specifics of your navigation or the way the elements are being mounted. One solution could be to move the `aria-live` attributes to the `checkedChildren` instead of the entire Choice component, so it will only announce when the `checkedChildren` is rendered. Another could be to attempt to add more logic to `aria-live` so it is only applied after the `input` value is changed by the user. @davidakennedy thoughts?

**Comment by davidakennedy** [comment]
Good suggestions, @bernardwang!

I'm hesitant to add an option to remove the ARIA live regions because that leaves the accessibility to be implemented elsewhere, and I'd rather things just work. So I like the idea of experimenting here to see if we can work around this.

> One solution could be to move the aria-live attributes to the checkedChildren instead of the entire Choice component, so it will only announce when the checkedChildren is rendered.

This is worth a try, and I can test the before and after to see how it works out.

> Another could be to attempt to add more logic to aria-live so it is only applied after the input value is changed by the user.

This may not work. I know that `aria-live` needs to be in the HTML source for the changes to be announced. So playing with the logic may mean it won't announce the things when it needs to announce them.

**Comment by bernardwang** [comment]
@hbillings Do you have more details on what code triggers this bug? Were the `ChoiceLists` initially rendered with a `checkedChildren` showing? If so, is the screenreader only reading the `Choice` content or does it also read the `checkedChildren` content? We are unable to reproduce this on our end so more information would help.

**Comment by forrestbaer** [comment]
Closing this issue here to be tracked in more recent issue #1055 since the codebase and team has shifted significantly since this request.
